graylog2-server/src/main/java/org/graylog2/indexer/rotation/strategies/TimeBasedSizeOptimizingStrategy.java

```diff
@@ -100,21 +105,25 @@ protected Result shouldRotate(final String index, IndexSet indexSet) {
     }
 
     private boolean indexExceedsLeeWay(DateTime creationDate, Period leeWay) {
-        final Instant now = Instant.now();
+        final Instant now = clock.instantNow();
         final Instant leewayLimit = now.minus(ROTATION_PERIOD.plus(leeWay));
 
-        return creationDate.isBefore(leewayLimit.toEpochMilli());
+        final Duration between = Duration.between(Instant.ofEpochMilli(creationDate.getMillis()), now);
+        return between.compareTo(Duration.ofDays(ROTATION_PERIOD.plus(leeWay).getDays())) >= 0;
+        //return creationDate.isBefore(leewayLimit.toEpochMilli());
```

Comment:
`leewayLimit` is now unused.
Can you add a comment briefly explaining the reasoning? It's hard to understand why date are compared this way. In fact I would break it out into a separate method, given that we use it twice. Then it can have a nice descriptive name.